src/iter.rs

```diff
@@ -57,11 +57,97 @@ where
     }
 }
 
+#[derive(Debug)]
+struct RingBuf<const SIZE: usize = 10> {
```

Comment:
It doesn't seem like making `SIZE` a generic const is helpful since we're not actually using that in this project?

Reply:
yeah that was to make playing with different sizes easier for me, i can take it out in the final commit.